apps/backend/src/controllers/booking.controller.ts

```
@@ -0,0 +1,75 @@
+    if (!userId) {
+      return res.status(401).json({
+        error: 'User authentication required'
+      });
+    }
+
+    // Log the payment confirmation attempt (without sensitive data in production)
+    if (process.env.NODE_ENV !== 'production') {
+      console.log('Payment confirmation attempt:', {
+        bookingId,
+        userId,
+        transactionHash: input.transactionHash.substring(0, 8) + '...'
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)